unlock-app/src/hooks/usePaywallLocks.ts

```diff
@@ -17,8 +20,14 @@ export const usePaywallLocks = (lockAddresses: string[]) => {
     })
 
     const locks = await Promise.all(lockPromises)
-    // getLock doesn't always return the lock address apparently
-    locks.forEach((lock, index) => (lock.address = lockAddresses[index]))
+    locks.forEach((lock, index) => {
+      if (lock.currencyContractAddress) {
+        getTokenBalance(lock.currencyContractAddress)
```

Comment:
We fire off a request for token balance whenever we run into an ERC20 lock. This can result in duplicate requests, but there should never be many of these.